src/relay/pass/simplify_bias_add.cc

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright (c) 2018 by Contributors
+ * \file expand_bias_add.cc
+ * \brief Expand bias_add to expand_dims and broadcast_add.
+ *        This can simplify the passes related to layout (e.g. alter_op_layout).
+ */
+#include <tvm/relay/pass.h>
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/attrs/nn.h>
+#include "pattern_util.h"
+
+namespace tvm {
+namespace relay {
+
+class BiasAddSimplifier : public ExprMutator {
```

Comment:
can we fold this into simplify inference?

Reply:
No. Because it is not only for inference.